static/app/components/organizations/pageFilters/container.tsx

```diff
@@ -148,6 +149,15 @@ function Container({
       return;
     }
 
+    // We may need to re-initialize the URL state if we completely clear
+    // out the global selection URL state, for example by navigating with
+    // the sidebar on the same view.
+    if (location.state?.source === SIDEBAR_NAVIGATION_SOURCE) {
+      doInitialization();
+      lastQuery.current = location.query;
+      return;
+    }
+
```

Comment:
Returning https://github.com/getsentry/sentry/pull/58634/commits/672455a6f1df0ddc0132bcf3cd2bc231c91a7223#r1372393326, but with a more explicit, targeted condition for re-initialization. This way, we don't reinitialize when user clicks "Reset".